File: app/src/main/java/se/wetcat/qatja/androidexample/MainActivity.java
package se.wetcat.qatja.androidexample;

/*
 * Copyright (C) 2017 Andreas Goransson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import android.os.Bundle;
import android.util.Log;

import se.wetcat.qatja.MQTTConnectionConstants;
import se.wetcat.qatja.androidexample.fragments.BrokerFragment;
import se.wetcat.qatja.androidexample.fragments.ChannelFragment;

import static se.wetcat.qatja.MQTTConnectionConstants.STATE_CONNECTED;
import static se.wetcat.qatja.MQTTConnectionConstants.STATE_NONE;

public class MainActivity extends ConnectionActivity
    implements BrokerFragment.OnBrokerFragmentChangedListener, ChannelFragment.OnChannelFragmentListener {

  private static final String TAG = MainActivity.class.getSimpleName();

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
  }

  @Override
  protected void onResume() {
    super.onResume();

//    if (getSupportFragmentManager().findFragmentByTag(BrokerFragment.class.getSimpleName()) == null) {
//      getSupportFragmentManager()
//          .beginTransaction()
//          .replace(R.id.container, BrokerFragment.newInstance(), BrokerFragment.class.getSimpleName())
//          .commit();
//    }
  }

  @Override
  public void onConnect(String host, String identifier) {
    connect(identifier, host);
  }

  @Override
  public void onStateChanged(int newState) {
    Log.d(TAG, "onStateChanged(" + MQTTConnectionConstants.resolveStateName(newState) + ")");


    if (newState == STATE_NONE) {
      if (getSupportFragmentManager().findFragmentByTag(BrokerFragment.class.getSimpleName()) == null) {
        getSupportFragmentManager()
            .beginTransaction()
            .replace(R.id.container, BrokerFragment.newInstance(), BrokerFragment.class.getSimpleName())
            .commit();
      }
    } else if (newState == STATE_CONNECTED) {
      if (getSupportFragmentManager().findFragmentByTag(ChannelFragment.class.getSimpleName()) == null) {
        getSupportFragmentManager()
            .beginTransaction()
            .replace(R.id.container, ChannelFragment.newInstance(), ChannelFragment.class.getSimpleName())
            .commit();
      }
    }
  }

  @Override
  public void onJoin(String channelName) {
    Log.e("ANTE", "onJoin(" + channelName + ")");

    subscribe(channelName);
  }
}

File: app/src/main/java/se/wetcat/qatja/androidexample/MainActivity.kt
package se.wetcat.qatja.androidexample

/*
 * Copyright (C) 2017 Andreas Goransson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import android.os.Bundle
import android.util.Log.d
import se.wetcat.qatja.androidexample.fragments.BrokerFragment
import se.wetcat.qatja.androidexample.fragments.ChannelFragment
import se.wetcat.qatja.androidexample.fragments.OnBrokerFragmentChangedListener
import se.wetcat.qatja.androidexample.fragments.OnChannelFragmentListener

class MainActivity : MqttActivity(), OnBrokerFragmentChangedListener, OnChannelFragmentListener {

    companion object {
        const val TAG = "MainActivity"
    }

    private var mEnteredHost: String? = null

    private var mEnteredIdentifier: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
    }

    override fun onResume() {
        super.onResume()

        if (supportFragmentManager.findFragmentByTag(BrokerFragment::class.java.simpleName) == null) {
            supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.container, BrokerFragment.newInstance(mEnteredHost, mEnteredIdentifier), BrokerFragment::class.java.simpleName)
                    .commit();
        }
    }

    override fun onConnect(host: String, identifier: String) {
        mEnteredHost = host
        mEnteredIdentifier = identifier

        attemptConnectMqtt(host, identifier)
    }

    override fun onJoin(channelName: String) {
        if (isDebug()) {
            d(TAG, "onJoin($channelName)")
        }

        attemptSubscribeToTopic(channelName)
    }

    override fun onMqttConnected() {
        if (supportFragmentManager.findFragmentByTag(ChannelFragment::class.java.simpleName) == null) {
            supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.container, ChannelFragment.newInstance(), ChannelFragment::class.java.simpleName)
                    .commit()
        }
    }

    override fun onMqttDisconnected() {
        if (supportFragmentManager.findFragmentByTag(BrokerFragment::class.java.simpleName) == null) {
            supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.container, BrokerFragment.newInstance(mEnteredHost, mEnteredIdentifier), BrokerFragment::class.java.simpleName)
                    .commit()
        }
    }

}
